src/components/ArticlePage/Comments/CommentForm.jsx

```diff
@@ -0,0 +1,54 @@
+import { Filter } from 'bad-words';
+import { UserContext } from '../../../contexts/User';
+import { useContext, useState } from 'react';
+import { postComment } from '../../../apiCalls';
+
+export const CommentForm = ({
+  article_id,
+  isPostingComment,
+  setIsPostingComment,
+  comments,
+  setComments,
+}) => {
+  const { user } = useContext(UserContext);
+  const [isMsgVisible, setIsMsgVisible] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const commentBody = e.target.form[0].value;
+    if (commentBody.length > 9) {
+      const filter = new Filter(); //remove bad words
+      postComment(filter.clean(commentBody), user.username, article_id).then(
+        (comment) => {
+          setTimeout(() => {
+            setComments([comment, ...comments]);
```

Comment:
Please revisit [these notes](l2c.northcoders.com/courses/sd-notes/front-end/#sectionId=react-state,step=mutating-state) for how to access current value of state and update non-primitive state.

This has the right idea though in principal 👍